### PR TITLE
add vesting pallet

### PR DESCRIPTION
**Pull request by enddynayn**
# Goal
The goal of this PR is 

Closes 

# Discussion

# Checklist
- [ ] Chain spec updated
- [ ] Updated the js/api-augment code if a custom RPC added/changed
- [ ] Design doc(s) updated
- [ ] Tests added
- [ ] Benchmarks added
- [ ] Weights updated

### CODE REVIEW

File: runtime/frequency/Cargo.toml [hunk 1]
```diff
@@ -137,10 +142,11 @@ std = [
   "xcm-builder/std",
   "xcm-executor/std",
   "xcm/std",
+  "orml-vesting/std", # "hex/std",
 ]
 
 runtime-benchmarks = [
-  "hex-literal",
+  # "hex-literal",
```
Comment (aramikm) [inline runtime/frequency/Cargo.toml:151]: Why was it required to comment this line?

File: runtime/frequency/src/lib.rs [hunk 2]
```diff
@@ -498,6 +506,87 @@ impl pallet_messages::Config for Runtime {
 	type MaxMessagePayloadSizeBytes = MaxMessagePayloadSizeBytes;
 }
 
+parameter_types! {
+	// Alice
+	// and Bob
+	pub TreasuryAccounts: Vec<AccountId> = vec![
+		hex_literal::hex!["d43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d"].into(),
```
Comment (aramikm) [inline runtime/frequency/src/lib.rs:521]: shouldn't this get configured from genesis?

File: runtime/frequency/src/lib.rs [hunk 3]
```diff
@@ -498,6 +506,87 @@ impl pallet_messages::Config for Runtime {
...
+		let zero_account_id = AccountId::decode(&mut sp_runtime::traits::TrailingZeroInput::zeroes())
+			.expect("infinite length input; no invalid inputs for type; qed");
+		Origin::from(RawOrigin::Signed(zero_account_id))
+	}
+} 
+
+parameter_types! {
+	// Alice
+	// and Bob
+	pub LibertyFoundationAccounts: Vec<AccountId> = vec![
```
Comment (aramikm) [inline runtime/frequency/src/lib.rs:555]: same here?

File: runtime/frequency/src/lib.rs [hunk 4]
```diff
@@ -518,6 +615,7 @@ construct_runtime!(
 		Timestamp: pallet_timestamp::{Pallet, Call, Storage, Inherent} = 2,
 		ParachainInfo: parachain_info::{Pallet, Storage, Config} = 3,
 		Sudo: pallet_sudo::{Pallet, Call, Config<T>, Storage, Event<T> }= 4,
+		// Utility: pallet_utility::{Pallet, Call, Event<T>}
```
Comment (aramikm) [inline runtime/frequency/src/lib.rs:618]: should this get removed if not necessary, usually commented code confuses others

**Review by aramikm — COMMENTED** [review]
I didn't read the whole logic in the pallet.
How much can be trust these libraries? I think we might need to understand all the logic to avoid unwanted consequences later.

**Comment by enddynayn** [comment]
@aramikm hey this is just me experimenting. This is not intended for merging.

**Pull request closed without merging**